drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -480,43 +496,45 @@ local function check_field_name_updates(device)
   end
 end
 
-local function assign_child_profile(device, child_ep)
+local function assign_switch_profile(device, switch_ep, is_child_device, electrical_tags)
   local profile
-
   for _, ep in ipairs(device.endpoints) do
-    if ep.endpoint_id == child_ep then
+    if ep.endpoint_id == switch_ep then
       -- Some devices report multiple device types which are a subset of
       -- a superset device type (For example, Dimmable Light is a superset of
       -- On/Off light). This mostly applies to the four light types, so we will want
       -- to match the profile for the superset device type. This can be done by
       -- matching to the device type with the highest ID
+      -- Note: Electrical Sensor does not follow the above logic, so it's ignored
       local id = 0
       for _, dt in ipairs(ep.device_types) do
-        id = math.max(id, dt.device_type_id)
+        if dt.device_type_id ~= ELECTRICAL_SENSOR_ID then
+          id = math.max(id, dt.device_type_id)
+        end
       end
       profile = device_type_profile_map[id]
       break
     end
   end
 
-  -- Check if device has an overridden child profile that differs from the profile that would match
-  -- the child's device type for the following two cases:
-  --   1. To add Electrical Sensor only to the first EDGE_CHILD (light-power-energy-powerConsumption)
-  --      for the Aqara Light Switch H2. The profile of the second EDGE_CHILD for this device is
-  --      determined in the "for" loop above (e.g., light-binary)
-  --   2. The selected profile for the child device matches the initial profile defined in
-  --      child_device_profile_overrides
-  for id, vendor in pairs(child_device_profile_overrides_per_vendor_id) do
-    for _, fingerprint in ipairs(vendor) do
-      if device.manufacturer_info.product_id == fingerprint.product_id and
-         ((device.manufacturer_info.vendor_id == AQARA_MANUFACTURER_ID and child_ep == 1) or profile == fingerprint.initial_profile) then
-         return fingerprint.target_profile
+  if electrical_tags ~= nil and (profile == "plug-binary" or profile == "plug-level" or profile == "light-binary") then
+    profile = string.gsub(profile, "-binary", "") .. electrical_tags
+  end
+
+  if is_child_device then
+    -- Check if child device has an overridden child profile that differs from the child's generic device type profile
+    for _, vendor in pairs(child_device_profile_overrides_per_vendor_id) do
+      for _, fingerprint in ipairs(vendor) do
+        if device.manufacturer_info.product_id == fingerprint.product_id and profile == fingerprint.initial_profile then
+          return fingerprint.target_profile
+        end
       end
     end
```

Comment:
```suggestion
  if is_child_device and child_device_profile_overrides_per_vendor_id[device.manufacturer_info.vendor_id] then
    -- Check if child device has an overridden child profile that differs from the child's generic device type profile
    for _, fingerprint in ipairs(child_device_profile_overrides_per_vendor_id[device.manufacturer_info.vendor_id]) do
      if device.manufacturer_info.product_id == fingerprint.product_id and profile == fingerprint.initial_profile then
        return fingerprint.target_profile
      end
    end
```
We could do something like this here to ensure the vendor id matches

Reply:
good idea